frontend/javascripts/oxalis/view/action-bar/starting_job_modals.tsx

```diff
@@ -788,9 +795,12 @@ export function MaterializeVolumeAnnotationModal({
           outputSegmentationLayerName,
         }) => {
           if (outputSegmentationLayerName == null) {
-            return Promise.resolve();
+            return;
           }
-          const volumeLayerName = getReadableNameOfVolumeLayer(segmentationLayer, tracing);
+          const volumeLayerName =
+            "fallbackLayer" in segmentationLayer && segmentationLayer.fallbackLayer != null
+              ? getReadableNameOfVolumeLayer(segmentationLayer, tracing)
+              : null;
```

Comment:
so, if no fallbackLayer exists `volumeLayerName` will simply be null? what will the job do then?

Reply:
> so, if no fallbackLayer exists volumeLayerName will simply be null? what will the job do then?

In short: Do not merge the fallback data with the volume annotation and simply take the volume data of the segmentation layer given and apply the skeletons of the merger mode.


Longer explanation (tried to make it clear :see_no_evil:):
The `volumeLayerName` is an optional parameter to the materialize job. The materialize job can apply merger mode skeletons as well as merging fallback data with the volume data given by a column annotation. 

In case the user has no volume annotation but e.g. wants its skeletons to be applied to a segmentation layer (without any volume annotation), the job just takes the volume layer's segmentation as the segmentation data. => segmentationLayer.fallbackLayer will be `null` and `getBaseSegmentationName` will return the layer name. Thus `volumeLayerName` is `null` and the job knows to not merge any segmentation data but to only apply the skeleton to the provided segmentation layer's data.

In case there is a volume annotation the fallback layer name still needs to be supplied so it will be merged with the volume annotation before a potential skeleton is applied to merge segments

I hope this is kinda clear. Maybe taking a quick look at the worker job also helps / explains it better: https://github.com/scalableminds/voxelytics/blob/06df1bb2785361b7bfe9abbccb1fb2027d0b2a65/voxelytics/worker/jobs/materialize_volume_annotation_legacy.py#L205

Reply:
Thanks for the clarification! It took me a while to grasp and I hope we can improve the naming and add some comments. I'll try to summarize (correct me if I'm wrong).

There are 3 cases:

1) There is a fallback segmentation and a volume annotation that is based on that. `volumeLayerName` will reference the volume layer. `baseSegmentationName` will reference the fallback segmentation. The worker job will merge those.
2) There is a segmentation layer and **no** volume annotation. `volumeLayerName` will be null (because `segmentationLayer.fallbackLayer`  will be null). `baseSegmentationName` will reference the only existing segmentation. The worker won't merge anything because `volumeLayerName` is null.
3) There is **no** fallback segmentation and **a** volume annotation. `volumeLayerName` will be null (because `segmentationLayer.fallbackLayer`  will be null). `baseSegmentationName` will reference the volume annotation. As in (2), no merging will be done.


I think, cases 1 + 2 make sense. However, in case 3, I find it quite confusing that the variable name `volumeLayerName` will be null even though a volume layer exists. Maybe you have an idea for clearing this up a bit. Changing the argument names could get a bit complicated because of the worker and the DB tables, right?
In the simplest case, add a comment about these three cases (like my enumeration above, if it is correct) :slightly_smiling_face:

Reply:
You are totally correct in your enumeration.

I originally designed the first version of this job afaik. I apologize for the confusing naming / confusing arguments passed to the job in case 3. I do not remember whether I had a good reason for this back then.

> In the simplest case, add a comment about these three cases (like my enumeration above, if it is correct) 🙂
Ok :+1: I did that